### ISSUE TITLE
sf project deploy start extremely slow compared to sfdx force:source:deploy

### ISSUE THREAD

**Issue by CollinHerber** [forcedotcom/cli#2129]
### Summary

When running `sf project deploy start -d /some/path` in a project it is taking on average 2.5x longer than the replaced `sfdx force:source:deploy -p /some/path`

Waiting for org pushes is already a large bottleneck in SF development and if it's now going to take 2.5x longer this will severely hamper efficiency.

### Steps To Reproduce:

1. Grab the latest of both the sfdx-cli and @salesforce/cli
2. Create a scratch org
3. Put a dozen or so files in a directory
4. Use the old source:deploy and project deploy commands.
5. Note the difference in time.

This has been replicated on every machine and org we have tested on at our workplace.
This is also present in CI environments as well.

### Expected result

The new command should be the same speed, if not faster.

### Actual result

The new command is slower.

### System Information

sfdx-cli version `7.199.7`
@salesforce/cli version `1.76.2`
Node version `18.16.0`
OS MacOSX `13.3.1`

**Comment by mshanemc** [comment]
if you don't need source tracking (which `source:deploy` doesn't do by default) you can opt out when you create the org

`sf org create scratch` has  `--no-track-source` 

That prevents the pre-deploy check for conflicts in the org, and prevents the org from having to wait for tracking information to be available after the deployment.

**Comment by CollinHerber** [comment]
That poses a problem because sometimes we want source tracking and sometimes we don't within the same org.

Take this for example. 
We have a repo with 5000+ files, which we have dozens of really, and with source tracking this goes through each of the directories in this org and sees if there are updates to change. This can take a really long time (45-60 seconds - Longer in some occasions). Let's say I am actively working within a single directory in this org and instead of waiting 45-60 seconds each time with a push, we can instead deploy a single directory. 

The problem then becomes if you need to pull in the latest changes from your main branch and need to push all of those new changes you would use the push command with does take longer because it looks for all file changes, but doesn't take as long as a fresh deploy.

TLDR sometimes we want source tracking and sometimes we don't within an org for efficiency sake.

My suggestion given that source tracking is always on with the deploy start is to allow the command to have a `--no-track-source` flag so it skips this step.

`sf project deploy start -d /some/path --no-track-source`

**Comment by yippie** [comment]
Yes please! Not having this (distinction between push and deploy) is a loss of very useful functionality.

**Comment by HyruleConsulting** [comment]
Hi there,
I also have this issue as I manage huge DX repositories for customers and the command is so slow it times out. Using the command as it is in CICD processes is not possible however using the command like this will work:

1. Run `sf project reset tracking`
2. Set Environment variable SF_DISABLE_SOURCE_MEMBER_POLLING=true
3. Run `sf project deploy start` and you'll find the command starts deploying straight away

The reason the command is slow is because it's querying the SourceMember table looking for conflicts prior to deploying/pushing.

**Comment by mshanemc** [comment]
The problem with those options (to do a deploy without tracking) is that it gets tracking in a completely messed up state.  Then you've got to keep in mind that the org is now not reliable if you want to push/pull etc again.

We got lots of complaints about this in the original commands ("I just wanted to deploy one file, so I did `source:deploy` and now I have conflicts").  It was a footgun.

@HyruleConsulting @yippie @CollinHerber 

How would it feel if it was an irrevocable process (ex: if you use `deploy start -d foo --no-track-source` we'd flip that Org's auth file to `tracksSource:false` so that it's not usable after tracking gets hosed?

**Comment by mshanemc** [comment]
@HyruleConsulting see above comments about turning off tracking from the `org create`. If you're doing CI/CI you probably don't want tracking operations on at all.

**Comment by HyruleConsulting** [comment]
@mshanemc 
Agree that for CICD purposes e.g. validating a commit - tracking source is not required. However if you have a huge a repository and a developer (or pipeline job) needs to build a scratch org from it (not talking about CICD processes here) then this command simply does not work that well for this scenario unless you do what I described (Reset tracking, disable source polling and then run `deploy start`).

Your suggestion of --no-track-source override flag on the `deploy start` command would work very well and I hope this feature is delivered soon so I can migrate customers from the old deployment commands to the new ones!

Thanks @mshanemc
